### DISCUSSION TITLE
How to send correct image content type header for static generated on nginx?

### DISCUSSION THREAD

**Discussion by pixelpaulaus** [nuxt/image#1127] [Q&A]
Hi All,
Seeing as NuxtIMG does not modify the filename extensions. If you serve the image via the server nuxt, it knows to send the correct content type header. However, if you generate a static site and serve it via something like nginx. it will get the desired header to send from the extension of the file (image.jpg). So instead of sending webp as the content type, it will send jpg as per the file extension.
So how does everyone else get the correct content type sent on nginx hosts?

**Comment by yudin-s** [comment]
For a fully static site, Nginx will normally choose `Content-Type` from the file extension, so a generated WebP payload stored under a `.jpg`-looking URL can indeed be served as `image/jpeg`.

You have a few practical options:

1. Configure Nginx to set the content type for the generated Nuxt Image/IPX format routes.
2. Avoid format conversion for static hosting unless the output path/extension can represent the converted format.
3. Pre-generate image files yourself with correct `.webp`/`.avif` filenames and reference those files directly.
4. Serve optimized images through a runtime image service instead of plain static files.

For Nginx, the fix is usually a targeted location rule for the generated image route/pattern, not changing global MIME behavior. For example, if all converted WebP files are under an `_ipx` URL containing `f_webp`, map that location to `default_type image/webp`.

The core issue is that static servers do not inspect image bytes like Nuxt's runtime server can; they rely on path/MIME config.
